nemo_run/core/execution/dgxcloud.py

```diff
@@ -90,29 +96,125 @@ def get_project_and_cluster_id(self, token: str) -> tuple[Optional[str], Optiona
+            with open(tarball_path, "rb") as file:
+                file_data = file.read()
+            encoded_data = base64.b64encode(file_data).decode("utf-8")
+
+            # Delete and recreate directory if it already exists, command to decode base64 data, save to a file, and extract inside the pod
+            cmd = f"rm -rf {dest_path} && mkdir -p {dest_path} && echo {encoded_data} | base64 -d > {dest_path}/archive.tar.gz && tar -xzf {dest_path}/archive.tar.gz -C {dest_path} && rm {dest_path}/archive.tar.gz"
+            return cmd
+
+        raise RuntimeError(f"Failed to generate data movement command")
+
+    def create_data_mover_workload(
+        self, token: str, project_id: str, cluster_id: str
+    ):
+        """
+        Creates an cpu only workload to move job directory into PVC using the provided project/cluster IDs.
+        """
+
+        cmd=self.copy_directory_data_command(self.job_dir, self.pvc_job_dir)
+
+        url = f"{self.base_url}/workloads/workspaces"
+        headers = self._default_headers(token=token)
+
+        payload = {
+            "name": "data-mover",
+            "useGivenNameAsPrefix": True,
+            "projectId": project_id,
+            "clusterId": cluster_id,
+            "spec": {
+                "command": "sh -c",
+                "args": f"'{cmd}'",
+                "image": "busybox:1.37.0",
+                "storage": {"pvc": self.pvcs},
+            }
+        }
+
+        response = requests.post(url, json=payload, headers=headers)
+
+        logger.debug(
+            "Created workload; response code=%s, content=%s",
+            response.status_code,
+            response.text.strip(),
+        )
+
+        return response
+
+    def delete_workload(
+            self, token: str, workload_id: str, sleep: int = 10
+    ):
+        url = f"{self.base_url}/workloads/workspaces/{workload_id}"
+        headers = self._default_headers(token=token)
+
+        response = requests.delete(url, headers=headers)
+
+        logger.debug(
+            "Delete interactive workspace; response code=%s, content=%s",
+            response.status_code,
+            response.text.strip(),
+        )
+        return response
+    
+    def move_data(
+            self, token: str, project_id: str, cluster_id: str, sleep: float = 10
+    ) -> None:
+        """
+            Moves job directory into PVC and deletes the workload after completion
+        """
+
+        resp = self.create_data_mover_workload(token, project_id, cluster_id)
+        if resp.status_code not in [200, 202]:
+            raise RuntimeError(f"Failed to create data mover workload, status_code={resp.status_code}")
+
+        resp_json = resp.json()
+        workload_id = resp_json["workloadId"]
+        status = DGXCloudState(resp_json["actualPhase"])
+
+        while status is DGXCloudState.PENDING or status is DGXCloudState.CREATING or status is DGXCloudState.INITIALIZING or status is DGXCloudState.RUNNING:
```

Comment:
Could also do `while status in [DGXCloudState.PENDING, DGXCloudState.CREATING, DGXCloudState.INITIALIZING, DGXCloudState.RUNNING]` potentially to clean this up a bit?